timemachine/cpp/src/context.cu

```diff
@@ -22,8 +22,16 @@ Context::Context(
     const double *box_0,
     std::shared_ptr<Integrator> intg,
     std::vector<std::shared_ptr<BoundPotential>> bps,
-    std::shared_ptr<MonteCarloBarostat> barostat)
-    : N_(N), barostat_(barostat), step_(0), d_sum_storage_(nullptr), d_sum_storage_bytes_(0), intg_(intg), bps_(bps) {
+    std::shared_ptr<MonteCarloBarostat> barostat,
+    bool freeze_reference)
```

Comment:
Do we want this to be a parameter in `Context` itself? Since for non-local MD use cases this parameter would be ignored?

Reply:
It defaults to true, so currently it behaves the same way as it use to.

The idea here is that we want to test these two cases (frozen/free reference) and get a sense of how the two behave. In the longer term, we will likely want to settle on a single implementation and this will go away for better version.

We could also add a separate function such as `ctxt.setup_local_md(temperature, frozen)` that can be called after construction to configure local md a specific way. And as I type this, I am inclined to think that is the way to go. And remove the `ensure_md_is_intialized` and only allow the user to call it once. Seems like a better way and could allow us to maintain both versions (if we wanted to) without additional jank.

Reply:
>We could also add a separate function such as ctxt.setup_local_md(temperature, frozen) that can be called after construction to configure local md a specific way. And as I type this, I am inclined to think that is the way to go. And remove the ensure_md_is_intialized and only allow the user to call it once. Seems like a better way and could allow us to maintain both versions (if we wanted to) without additional jank.

💯

Reply:
Reworked in https://github.com/proteneer/timemachine/pull/1014/commits/39087381d71a0f57369c6b08ac785e0442eb19b0